src/Filters/WhereSearch.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/*
+ * Copyright 2022 Cloud Creativity Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+declare(strict_types=1);
+
+namespace LaravelJsonApi\Eloquent\Filters;
+
+use LaravelJsonApi\Eloquent\Contracts\Filter;
+use LaravelJsonApi\Eloquent\Filters\Concerns\DeserializesValue;
+
+class WhereSearch implements Filter
+{
+    use DeserializesValue;
+    use Concerns\IsSingular;
+
+    /**
+     * @var string
+     */
+    private string $name;
+
+    /**
+     * Create a new filter.
+     *
+     * @param string $name
+     * @param string|null $column
+     * @return static
+     */
+    public static function make(string $name, string $columns): self
+    {
+        return new static($name, $columns);
+    }
+
+    /**
+     * WhereSearch constructor.
+     *
+     * @param string $name
+     * @param string $column
+     */
+    public function __construct(string $name, string $columns)
```

Comment:
I really like the idea of the filter supporting multiple columns. There's a couple of improvements here though for both the `make()` and `__construct()` methods.

Let's change the parameters so that the signature is:

```php
/**
 * WhereLike constructor.
 * 
 * @param string $name
 * @param string|string[] $columns
 */
 public function __construct(string $name, $columns = [])
 {
    $this->name = $name;
    $this->columns = empty($columns) ? [$name] : Arr::wrap($columns);
 }
```

I.e. the idea is:

1. As with other filters, if no column is specified we assume the column is the same as `$name`.
2. We support one column, using `Arr::wrap` to turn it into an array of columns.
3. We support multiple columns, provided as an array.

Reply:
You'll need to add a `private array $columns` property, as the class is currently missing one.